stack.py

```diff
@@ -0,0 +1,29 @@
+class StackItem(object):
+    def __init__(self, data, next_item=None):
+        self.data = data
+        self.next_item = next_item
+
+    def __str__(self):
+        return self.data
+
+
+class StackFrame(object):
+    def __init__(self, first_item=None):
+        self.first_item = first_item
+
+    def push(self, new_item):
```

Comment:
values in, values out.  We should not have to deal with `StackItems` at all outside the stack.